### DISCUSSION TITLE
Device transfer

### DISCUSSION THREAD

**Discussion by BreadDain** [IHateMyKite/UnforgivingDevices#83] [Show and tell]
Decided to spend some time on a script that is capable of transfering devices from one actor to another while keeping all modifiers intact, the [result is pushed to a separate branch](https://github.com/IHateMyKite/UnforgivingDevices/tree/device-transfer-script). It sort of works with some crutches, but is extremely slow. IHateMyKite, if you think it might become useful somewhere - feel free to grab anything from there. Plugin with set up magic effect and spell is included.

**Comment by IHateMyKite** [comment]
Interesting idea. But I thing that there is much simpler way of transferring device. It is already to some extent implemented, but not for actor. Basically, you can use RemoveItem function (godlike function which is the main reason that NPC support was even possible). There is a super secret last argument. If you actually put some object here (like actor or container), it will transfer the item to that container. The transferred item will be no copy, but actually the same item. So you can transfer the item to the actor this way and that just equip it. Only thing you need to do is to also change Wearer variable on device and the device (which might be the hardest part of this implementation, as you need to get the script). You will also have to check that actor has no other objects in inventory, as it's possible that if actor have more than 1 of the same device, it will not work.

**Comment by BreadDain** [comment]
That would actually be way faster. Although I recently came to conclusion that anything related to inventory device enumeration has to be done with asynchronous calls to reduce execution time. Big inventories or large amount of devices take way too long to process.

**Comment by BreadDain** [comment]
Redid the [script](https://github.com/IHateMyKite/UnforgivingDevices/blob/device-transfer-script/Scripts/Source/UD_CompassionMagEffect_Script.psc) entirely, it is way faster indeed, but I have a haunting feeling that there is something I've missed. Not only I had issues with NPC outfit/calm spell staying on, but also there is too many variables changing during LockDevice that I might have missed some of them.

Had to add a setter interface to [CustomDevice_RenderScript to make it work.](https://github.com/IHateMyKite/UnforgivingDevices/blob/c5b58c554797b703565b7661d528f0b0fd171724/Scripts/Source/UD_CustomDevice_RenderScript.psc#L1124-L1129)
